pkg/realserver/configure.go

```diff
@@ -338,6 +379,96 @@ func (r *realserver) periodic() error {
 	}
 }
 
+// ConfigureHAProxy this function is the bridge between a v6 address and a v4
+// pod address. This function iterates over the declared v6 configs and backends
+// for each, checks if any pods match that service selector, and creates a
+// haproxy instance for each backend that maps the VIP:PORT to a list of backend
+// these are the pod ips, not the service IPs, to ensure traffic stays on-node
+func (r *realserver) ConfigureHAProxy() error {
+
+	configSet := []haproxy.VIPConfig{}
+	for ip, config := range r.config.Config6 {
+		// make a single haproxy server for each v6 VIP with all backends
+		for port, service := range config {
+			// fetch the service config and pluck the clusterIP
+			if !r.node.HasServiceRunning(service.Namespace, service.Service, service.PortName) {
+				r.logger.Warnf("no service found for configuration [%s]:(%s/%s), skipping haproxy config", string(ip), service.Namespace, service.Service)
+				continue
+			}
+
+			ips := r.node.GetPodIPs(service.Namespace, service.Service, service.PortName)
+
+			services := r.watcher.Services()
+			serviceName := fmt.Sprintf("%s/%s", service.Namespace, service.Service)
+			serviceForConfig := services[serviceName]
+			if service == nil {
+				return fmt.Errorf("error creating haproxy configs. Could not find kube service %s on ip [%s]", serviceName, string(ip))
+			}
+
+			// iterate over service ports and retrieve the one we want for this config
+			// we search for targetPort, the actual port open on the pod IP on this node
+			var targetPortForService string
+			for _, servicePort := range serviceForConfig.Spec.Ports {
+				if service.Service == serviceForConfig.Name {
+					/*
+						this is an annoying kube type IntOrString, so we have to
+						decide which it is and then set it as uint16 here
+						additionally if targetport is not defined, targetPort == port:
+
+						kube docs: "Note: A Service can map any incoming port to a targetPort.
+						By default and for convenience, the targetPort is set to the same value as the port field."
+
+						This case is our third check here
+					*/
+					if servicePort.TargetPort.StrVal != "" {
+						targetPortForService = servicePort.TargetPort.StrVal
+					} else if servicePort.TargetPort.IntVal != 0 {
+						targetPortForService = strconv.Itoa(int(servicePort.TargetPort.IntVal))
+					} else {
+						// targetPort == port
+						targetPortForService = string(servicePort.Port)
+					}
```

Comment:
should this break?  why iterate over the rest of the ports? if there are >1 ports on the service which one should it pick? how would we test this function?

Reply:
"if there are  > 1 ports": This is parsed further upstream, by the user providing the service name, port, namespace, in their config

```for _, servicePort := range serviceForConfig.Spec.Ports {
	if service.Service == serviceForConfig.Name {```

However it should break, which I added